docs/react/guides/suspense.md

```diff
@@ -98,10 +96,53 @@ const App: React.FC = () => {
 }
 ```
 
+## useSuspenseQuery
+
+You can also use the dedicated `useSuspenseQuery` hook to enable suspense mode for a query:
+
+```tsx
+import { useSuspenseQuery } from '@tanstack/react-query'
+
+const { data } = useSuspenseQuery({ queryKey, queryFn })
+```
+
+This has the same effect as setting the `suspense` option to `true` in the query config, but it works better in TypeScript, because `data` is guaranteed to be defined (as errors and loading states are handled by Suspense- and ErrorBoundaries).
+
+On the flip side, you therefore can't conditionally enable / disable the Query. `placeholderData` also doesn't exist for this Query. To prevent the UI from being replaced by a fallback during an update, wrap your updates that change the QueryKey into [startTransition](https://react.dev/reference/react/Suspense#preventing-unwanted-fallbacks).
```

Comment:
I have a few questions in this description.
Could you explain why `placeholderData` is missing from the `useSuspenseQuery` option?
I'm a little confused as to what the suspense mode and placeholderData options are related to.

Reply:
I don't have much context, i guess because placeholder is only used when state is 'pending', which with useSuspenseQuery will suspend the current component and thus won't be used. Under the hood it os passed to a code-path where it is supported, but it is just ignored in this particular case. But @TkDodo will have an accurate answer

Reply:
@incepter is right, that's the reason. `placeholderData` is meant to be shown while you are initially fetching. With suspense, that is what `Suspense` boundaries and fallbacks can do.

Also, in v5, we are using `placeholderData: (previous) => previous` as a replacement for `keepPreviousData`. It allows us to show data of a previous query key while the new query key is fetching for the first time. This is useful for e.g. pagination.

With suspense, this is what `transitions` are doing: wrap the thing that triggers the change for the query key into a `startTransition` to get the same behaviour, so you don't need `placeholderData`. It also doesn't work well with suspense, see this issue:

- https://github.com/TanStack/query/issues/3572

Reply:
https://github.com/TanStack/query/issues/3572#issuecomment-1304749251
> `keepPreviousData` becomes unnecessary when you use `suspense` + `startTransition`.
> ...
> `keepPreviousData` is meant to give you the "suspense" experience without actually using suspense.

Thanks! @TkDodo 
I understand why you removed the `placeholderData` option.